### PR TITLE
check value correctly before transfer

### PR DESCRIPTION
**Pull request by ec2**
**Summary of changes**
Changes introduced in this pull request:
- tied for worlds smallest PR

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

File: vm/interpreter/src/default_runtime.rs [hunk 1]
```diff
@@ -660,7 +661,7 @@ where
             .on_method_invocation(msg.value(), msg.method_num()),
     )?;
 
-    if msg.value() > &TokenAmount::from(0) {
+    if !msg.value().is_zero() {
```
Comment (austinabell) [inline vm/interpreter/src/default_runtime.rs:664]: this value could be negative, might be better to check 
```
msg.sign() != Sign::Minus
```

Comment (austinabell) [inline vm/interpreter/src/default_runtime.rs:664]: Wait, why can this be negative?

Comment (ec2) [reply vm/interpreter/src/default_runtime.rs:664]: That is handled in the transfer function

Comment (austinabell) [reply vm/interpreter/src/default_runtime.rs:664]: oh yeah, that's super weird, was wondering about that failed test case

**Pull request merged**